src/Microsoft.OData.Core/ODataParameterReaderCore.cs

```diff
@@ -280,9 +280,14 @@ protected internal void EnterScope(ODataParameterReaderState state, string name,
                     // Note that the binding parameter will be specified on the Uri rather than the payload, skip the binding parameter.
                     foreach (IEdmOperationParameter parameter in this.Operation.Parameters.Skip(this.Operation.IsBound ? 1 : 0))
                     {
-                        if (!this.parametersRead.Contains(parameter.Name) && !this.inputContext.EdmTypeResolver.GetParameterType(parameter).IsNullable)
+                        bool isOptionalPameter = parameter is IEdmOptionalParameter;
+                        if (!isOptionalPameter)
                         {
-                            missingParameters.Add(parameter.Name);
+                            // only check the non-optional parameter
+                            if (!this.parametersRead.Contains(parameter.Name) && !this.inputContext.EdmTypeResolver.GetParameterType(parameter).IsNullable)

```

Comment:
Combine !isOptionalParameter in this statement?

Reply:
+1